src/components/utils/src/rwlock_win.cc

```diff
@@ -69,82 +65,65 @@ sync_primitives::RWLock::~RWLock() {
   delete impl_;
 }
 
-bool sync_primitives::RWLock::AcquireForReading() {
-  return impl_->AcquireForReading();
+void sync_primitives::RWLock::AcquireForReading() {
+  impl_->AcquireForReading();
 }
 
 bool sync_primitives::RWLock::TryAcquireForReading() {
   return impl_->TryAcquireForReading();
 }
 
-bool sync_primitives::RWLock::AcquireForWriting() {
-  return impl_->AcquireForWriting();
+void sync_primitives::RWLock::AcquireForWriting() {
+  impl_->AcquireForWriting();
 }
 
 bool sync_primitives::RWLock::TryAcquireForWriting() {
   return impl_->TryAcquireForWriting();
 }
 
-bool sync_primitives::RWLock::Release() {
-  return impl_->Release();
+void sync_primitives::RWLock::ReleaseForReading() {
+  impl_->ReleaseForReading();
+}
+
+void sync_primitives::RWLock::ReleaseForWriting() {
+  impl_->ReleaseForWriting();
 }
 
-sync_primitives::RWLock::Impl::Impl()
-  : status_(kNotAcquired) {
+sync_primitives::RWLock::Impl::Impl() {
   InitializeSRWLock(&rwlock_);
 }
 
 sync_primitives::RWLock::Impl::~Impl() {
-  sync_primitives::AutoLock lock(status_lock_);
-  if (kNotAcquired != status_) {
-    LOG4CXX_ERROR(logger_, "RWLock is acquired");
-  }
 }
 
-bool sync_primitives::RWLock::Impl::AcquireForReading() {
-  sync_primitives::AutoLock lock(status_lock_);
+void sync_primitives::RWLock::Impl::AcquireForReading() {
   AcquireSRWLockShared(&rwlock_);
-  status_ = kAcquiredForReading;
-  return true;
 }
 
 bool sync_primitives::RWLock::Impl::TryAcquireForReading() {
-  sync_primitives::AutoLock lock(status_lock_);
   if (!TryAcquireSRWLockShared(&rwlock_)) {
     LOG4CXX_WARN(logger_, "Failed to acquire rwlock for reading");
     return false;
   }
-  status_ = kAcquiredForReading;
   return true;
 }
 
-bool sync_primitives::RWLock::Impl::AcquireForWriting() {
-  sync_primitives::AutoLock lock(status_lock_);
+void sync_primitives::RWLock::Impl::AcquireForWriting() {
   AcquireSRWLockExclusive(&rwlock_);
-  status_ = kAcquiredForWriting;
-  return true;
 }
 
 bool sync_primitives::RWLock::Impl::TryAcquireForWriting() {
-  sync_primitives::AutoLock lock(status_lock_);
   if (!TryAcquireSRWLockExclusive(&rwlock_)) {
     LOG4CXX_WARN(logger_, "Failed to acquire rwlock for writing");
```

Comment:
@pestOO, please, describe your vision of this situation. this PR is already merged. how we should react on your comment in such case?